### PR TITLE
Feature/info box

### PR DESCRIPTION
**Pull request by tieswesterborg**
Infobox with info and show other 'zoom levels'

### CODE REVIEW

File: src/main/java/application/controllers/GraphMouseHandling.java [hunk 1]
```diff
@@ -11,22 +11,27 @@
...
-    };*/
 
+        core.Node clicked = mainController.getGraphController().getGraph()
+                .getModel().getLevelMaps().get(0).get(node.getCellId());
 
-    EventHandler<MouseEvent> onMouseReleasedEventHandler = event -> {
+        System.out.println(mainController.getGraphController().getGraph()
+                .getModel().getLevelMaps().size());
 
+        String info = "";
```
Comment (ghost) [inline src/main/java/application/controllers/GraphMouseHandling.java:28]: Using a StringBuilder might be neater here.

File: src/main/java/application/controllers/MainController.java [hunk 2]
```diff
@@ -56,41 +67,70 @@ public final void initialize(URL location, ResourceBundle resources) {
...
+    private void createInfoList(String info) {
+        listVBox = new VBox();
+        infoScroller = new ScrollPane();
+
+        infoScroller.setHbarPolicy(ScrollPane.ScrollBarPolicy.NEVER);
+        infoScroller.setVbarPolicy(ScrollPane.ScrollBarPolicy.NEVER);
+        infoScroller.prefHeightProperty().bind(listVBox.heightProperty());
+        infoScroller.prefWidth(screenSize.getWidth() / 5);
+
+        if (info == "") {
```
Comment (ghost) [inline src/main/java/application/controllers/MainController.java]: I would use isEmpty() here

File: src/main/java/application/controllers/MainController.java [hunk 3]
```diff
@@ -56,41 +67,70 @@ public final void initialize(URL location, ResourceBundle resources) {
...
+
+        infoList.getChildren().addAll(id);
+    }
+
+    /**
+     * Modify the information of the Node.
+     *
+     * @param iD desired info.
+     */
+    public void modifyNodeInfo(String iD) {
```
Comment (NielsW-dev) [inline src/main/java/application/controllers/MainController.java]: Replace "iD" with "id"

File: src/main/java/application/controllers/MainController.java [hunk 4]
```diff
@@ -132,9 +179,16 @@ public void createMenu() {
...
      * Switches the scene to the graph view.
+     *
+     * @param delta the diff in view to apply.
      */
-    public void switchScene() {
+    public void switchScene(int delta) {
+        currentView += delta;
+        currentView = Math.max(0, currentView);
+
+        currentView = Math.min(9, currentView);
```
Comment (ghost) [inline src/main/java/application/controllers/MainController.java:189]: Please add a comment that explains what happens with the altering of 'currentView'.

File: src/main/java/core/graph/Graph.java [hunk 5]
```diff
@@ -45,22 +45,31 @@ public Model getModel() {
...
-        HashMap<Integer, Node> nodeMap = levelMaps.get(levelMaps.size() - 1);
+        model.setLevelMaps(levelMaps);
+
+        if (depth > levelMaps.size() - 1) {
+            depth = levelMaps.size() - 1;
+        } else if (depth < 0) {
+            depth = 0;
+        }
+        HashMap<Integer, Node> nodeMap = levelMaps.get(depth);
+        System.out.println("Loading map: " + depth);
```
Comment (ghost) [inline src/main/java/core/graph/Graph.java:76]: Appears to be a debug comment, should be removed if the code is tested.

File: src/main/java/core/GraphReducer.java [hunk 6]
```diff
@@ -61,12 +68,34 @@ public static void determineParents(HashMap<Integer, Node> nodeMap) {
...
+            Node n = map.get(i);
+            Node newNode = new Node(n.getId(), n.getSequence(), n.getzIndex());
+            newNode.setLinks(n.getLinks());
+            newNode.setParents(n.getParents());
+            newNode.setGenomes(n.getGenomes());
+            res.put(i, newNode);
+        }
+
+        return res;
+    }
```
Comment (ric-jon) [inline src/main/java/core/GraphReducer.java:91]: Does HashMap.clone() not suffice?

Comment (ghost) [inline src/main/java/core/GraphReducer.java:91]: No, in that case only a shallow copy will be made and level map 0 will have collapsed node, something that should not happen.

File: src/main/java/core/Parser.java [hunk 7]
```diff
@@ -14,62 +14,77 @@
...
+    /**
+     * Read the data from a .gfa file and put the nodes in a hashmap.
+     *
+     * @param input - filepath of .gfa file to be parsed.
+     * @return - A HashMap containing the information from the .gfa file.
+     * @throws IOException Throw exception on read GFA read failure.
+     */
+    @SuppressWarnings("checkstyle:magicnumbers")
+    @SuppressFBWarnings("I18N")
+    public final HashMap<Integer, Node>
```
Comment (ghost) [inline src/main/java/core/Parser.java:38]: Method is too long, needs refactoring

**Pull request merged**